test/widget/auth/email_auth_options_page_test.dart

```diff
@@ -329,50 +331,93 @@ void main() {
     });
 
     testWidgets(
-        'After input valid email/password/repeatPassword, createAccountButton dispatches SignUpWithEmail action',
+        'CreateAccountButton only dispatches when all textfields are valid',
         (WidgetTester tester) async {
-      // Setup the app state with expected values
-      final initialAppState = AppState.init();
-      final alteredState = initialAppState
-          .rebuild((b) => b..emailAuthOptionsPage.mode = EmailAuthMode.signUp);
-      final testMiddleware = VerifyDispatchMiddleware();
-
-      // Create the test harness.
-      final store = Store<AppState>(appReducer,
-          initialState: alteredState, middleware: [testMiddleware]);
+      // Setup a fake store and test harness with intial values.
+      final fakeStore = FakeStore(
+          updates: (b) => b..emailAuthOptionsPage.mode = EmailAuthMode.signUp);
       final wut = EmailAuthOptionsPage();
-      final harness =
-          StoreProvider<AppState>(store: store, child: MaterialApp(home: wut));
+      final harness = StoreProvider<AppState>(
+          store: fakeStore, child: MaterialApp(home: wut));
 
       // Tell the tester to build the widget tree.
       await tester.pumpWidget(harness);
 
-      // Create the Finders.
+      // Create Finders for text fields and create account button.
       final createAccountButton = find.byType(CreateAccountButton);
-      expect(createAccountButton, findsOneWidget);
       final emailTextField = find.byType(EmailTextField);
-      expect(emailTextField, findsOneWidget);
       final passwordTextField = find.byType(PasswordTextField);
-      expect(passwordTextField, findsOneWidget);
       final repeatPasswordTextField = find.byType(RepeatPasswordTextField);
+
+      // Check that all the expected widgets are present.
+      expect(createAccountButton, findsOneWidget);
+      expect(emailTextField, findsOneWidget);
+      expect(passwordTextField, findsOneWidget);
       expect(repeatPasswordTextField, findsOneWidget);
 
-      // Tap to validate form
+      // Create Finders for the text shown by the textfields when invalid.
+      final invalidEmailText = find.text('please enter a valid email');
```

Comment:
Added some extra checks while I was debugging and thought they might as well stay in.

Reply:
noice